proposed/vector-instruction-set-default.md

```diff
@@ -0,0 +1,100 @@
+
+SSE4 was introduced with the [Intel Core 2](https://en.wikipedia.org/wiki/Intel_Core_(microarchitecture)) launch in 2006, along with x64. That means that all x64 chips are SSE4 capable, and that SSE2 is really just a holdover from our 32-bit computer heritage. That would make SSE2 a poor baseline for x64 software, but a good one for 32-bit.
+
+AVX2 was released in late 2013 with [Intel Haswell](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX2) processors. After more than seven years in market, we should be able to take advantage of AVX2 more actively, to the end of improved performance, and lighting up even more CPU transistors.
+
+The advantage of AVX2 is not tens of percentage points. It is important not to oversell the importance of using later SIMD instructions. In the general case, using AVX2 might deliver low double-digit wins. For server apps that run all day, this could be important, and might allow for using a lower-cost tier of cloud machine and/or delivering higher performance. We also expect a significant startup improvement.
+
+You might be wondering about [AVX-512](https://en.wikipedia.org/wiki/AVX-512). Hardware intrinsics have not yet been defined (for .NET) for AVX-512. AVX-512 is also known to have more [nuanced performance](https://lemire.me/blog/2018/08/25/avx-512-throttling-heavy-instructions-are-maybe-not-so-dangerous/). For now, our vectorized code tops out at AVX2 for the x86-x64 [ISA](https://en.wikipedia.org/wiki/Instruction_set_architecture). Also, too few machines support AVX-512 to consider making it a default choice.
+
+You might be wondering about Arm processors. Assuming we adopted this plan for x86-x64, we'd do something similar for Arm64. The Arm ISA defines [NEON](https://en.wikipedia.org/wiki/ARM_architecture#Advanced_SIMD_(Neon)) as part of [Armv7](https://en.wikipedia.org/wiki/ARM_architecture#32-bit_architecture), NEON (enhanced) in [Armv8A](https://en.wikipedia.org/wiki/AArch64#AArch64_features) and [SVE](https://en.wikipedia.org/wiki/AArch64#Scalable_Vector_Extension_(SVE)) as part of [Armv8.2A](https://en.wikipedia.org/wiki/AArch64#ARMv8.2-A). [SVE2](https://en.wikipedia.org/wiki/AArch64#Scalable_Vector_Extension_2_(SVE2)) will appear sometime in the future. The .NET code base isn't vectorized much or at all for Arm32, such that this proposal doesn't apply for 32-bit Arm. On Arm64, we'd need to ensure that we make choices that take Raspberry Pi, laptops and any other form factors into consideration.
```

Comment:
For ARM64, we currently consider `System.Runtime.Intrinsics.Arm.AdvSimd` as a "baseline" instruction set and is required by any ARM64 chip running .NET

Reply:
Is Advanced SIMD the same thing as "advanced NEON", as opposed to SVE/SVE2?

Reply:
Yes, it should be. AFAIK, `NEON` is the branded name for when it was introduced on ARM32 chips, while `AdvSIMD` is the name used for ARM64 machines and in the architecture manual.

There are some minor differences between them, but they are largely compatible and we designed the `System.Runtime.Intrinsics.Arm` namespace with that in mind (with input from the folks at ARM).

Reply:
I don't know. Wikipedia article refer to Advanced SIMD as NEON in two places:

- https://en.wikipedia.org/wiki/AArch64#AArch64_features
- https://en.wikipedia.org/wiki/ARM_architecture#Advanced_SIMD_(NEON)

I have no horse in this race. Just sharing what I read.